learning_resources.md

```diff
@@ -162,3 +167,8 @@
 * [Latency Numbers Every Engineer Should Know](https://quizlet.com/91957128/latency-numbers-every-programmer-should-know-flash-cards/)  
   Flash cards with useful estimates for reasoning about how long operations
   take.
+
+### Functional Programming
+* [Haskell Programming from First Principles](http://haskellbook.com)
```

Comment:
Please add that the first chapter is offered as a free sample, and list the price of the book.